Error while multiplying a `SupportsInt` and an int

I've got the following code:
``` python
from decimal import Decimal
d = Decimal('10')
x = max(d, 0) * 3
```
Mypy types `max(d, 0)` as a SupportsInt, which seems correct, but then complains:
```
test.py:3: error: Unsupported operand types for * ("SupportsInt" and "int")
```
I expect to be able to perform any operation between X and SupportsX if that operation is supported between X and another X.

This is mypy 0.570 and the underlying code is Python 3.6.1. I can reproduce this issue with no flags passed.

--- a/docs/source/class_basics.rst
+++ b/docs/source/class_basics.rst
@@ -312,7 +312,8 @@ This is a type for objects that support ``bytes(x)``.
 ``SupportsComplex``
 -------------------
 
-This is a type for objects that support ``complex(x)``.
+This is a type for objects that support ``complex(x)``. Note that no arithmetic operations
+are supported.
 
 .. code-block:: python
 
@@ -321,7 +322,8 @@ This is a type for objects that support ``complex(x)``.
 ``SupportsFloat``
 -----------------
 
-This is a type for objects that support ``float(x)``.
+This is a type for objects that support ``float(x)``. Note that no arithmetic operations
+are supported.
 
 .. code-block:: python
 
@@ -330,7 +332,8 @@ This is a type for objects that support ``float(x)``.
 ``SupportsInt``
 ---------------
 
-This is a type for objects that support ``int(x)``.
+This is a type for objects that support ``int(x)``.  Note that no arithmetic operations
+are supported.
 
 .. code-block:: python
 

